shared/src/metabase/mbql/util.cljc

```diff
@@ -455,7 +455,10 @@
   "Is a given artihmetics clause operating on datetimes?"
   [clause]
   (mbql.match/match-one clause
-    #{:interval :relative-datetime}
+    #{:datediff :temporal-extract}
```

Comment:
This function is used to determine what `:+` is supposed to do in the query processor, either addition time to a date or the more traditional addition of numbers. But now we have datetimes that appear in arithmetic expressions `[:+ 3 [:datediff $created_at $shipped_at :hours]]` is a valid addition expression and should not identify the `{:temporal-unit :default}` of the timestamp columns to mean the whole thing is adding dates together.

Reply:
It's also used here in infer-expression-type, to identify expressions that return DateTime types. `https://github.com/metabase/metabase/blob/16ed4b0837580e21126006a06ec57d167d46b654/src/metabase/query_processor/middleware/annotate.clj#L162`